mllib/src/main/scala/org/apache/spark/ml/clustering/PowerIterationClustering.scala

```diff
@@ -232,7 +232,7 @@ class PowerIterationClustering private[clustering] (
         case _: LongType =>
           uncastPredictions
         case otherType =>
```

Comment:
Why not directly use
`case intType: IntType: ` so that make a stronger restriction ?
Or do we need to support other types besides int/long ?